algebird-core/src/main/scala/com/twitter/algebird/CountMinSketch.scala

```diff
@@ -578,8 +652,19 @@ case class CMSInstance[K](countsTable: CMSInstance.CountsTable[K],
     }
   }
 
-  def frequency(item: K): Approximate[Long] =
-    makeApprox(countsTable.counts.iterator.zip(params.hashes.iterator).map { case (row, hash) => row(hash(item)) }.min)
+  def frequency(item: K): Approximate[Long] = {
+    var freq = Long.MaxValue
+    val hs = params.hashes
+    val it = countsTable.counts.iterator
+    var i = 0
+    while (it.hasNext) {
```

Comment:
in the case of `freq == 0` we could stop, since we know that `freq >= 0` and `min` can only decrease it. It will only help in the case of missing keys so it might not be worth adding the `while(it.hasNext && (freq > 0))` which might slightly slow down non-zero keys.